File: Data Structures & Algorithms/is-anagram/submission-3.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {
        if (s.length !== t.length) {
            return false
        }

        let charCount = {}

        for (let i = 0; i < s.length; i++) {
            charCount[s[i]] = (charCount[s[i]] || 0) + 1;
            charCount[t[i]] = (charCount[t[i]] || 0) - 1;

        }

        console.log(charCount)

        for (let count in charCount) {
            if (charCount[count] !== 0) {
                return false
            }
        }

        return true
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-3.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        # Largest pile of bananas
        left = 1 # Min Speed
        right = max(piles)
        
        while left < right:
            count = 0
            # K becomes midpoint
        
            k = math.floor((left+right)/2)

            for bananas in piles:
                count = count + math.ceil(bananas / k)
            
            if count > h:
                left = k + 1
            else:
                right = k

        return int(left)
        
        
